opensrp-opd/src/main/java/org/smartregister/opd/OpdLibrary.java

```diff
@@ -309,8 +309,11 @@ public Event processOpdCheckInForm(@NonNull String eventType, String jsonString,
 
     public List<Event> processOpdDiagnosisAndTreatmentForm(@NonNull String jsonString, @NonNull Intent data) throws JSONException {
         JSONObject jsonFormObject = new JSONObject(jsonString);
+
         JSONObject step1JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP1);
 
+        int repeatedGroupNum = OpdUtils.buildRepeatingGroupTests(step1JsonObject);
+
         JSONObject step2JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP2);
 
         JSONObject step3JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP3);
```

Comment:
Please update this to be able to work with more than 3 steps.  Have a default function for things that don't need any custom processing.